### PR TITLE
Fixes for  `mov` op for nz assembler

### PR DESCRIPTION
**Pull request by Svenito**
A number of fixes and updates to the `mov` command for the _nz_ assembler including:

* handling of `esp` register in more cases
* able to handle negative offsets for target register both byte and word
* able to handle negative offsets for target with immediate value as
        source, both byte and word
* refactor mov block of code to be more consistent (further work needed)

Goes with r2r PR radare/radare2-regressions#435

### CODE REVIEW

File: libr/asm/p/asm_x86_nz.c [hunk 1]
```diff
@@ -1141,26 +1142,48 @@ SETNP/SETPO - Set if No Parity / Set if Parity Odd (386+)
...
 				if (delta) {
 					*delta++ = 0;
 				} else {
 					delta = strchr (arg, '-');
 					if (delta) {
-						ut32 n = -getnum (a, delta + 1);
-						ut8 *N = (ut8 *)&n;
+						int r = getreg (arg);
+						int r2;
+						int r2_is_num = 0;
```
Comment (radare) [inline libr/asm/p/asm_x86_nz.c]: this is `bool` , right?

File: libr/asm/p/asm_x86_nz.c [hunk 2]
```diff
@@ -1254,13 +1267,12 @@ SETNP/SETPO - Set if No Parity / Set if Parity Odd (386+)
...
 				} else {
 					// mov [num], reg
-					ut8 *ptr = (ut8 *)&num;
 					data[l++] = 0x89;
 					data[l++] = (r0 << 3) | 5;
-					data[l++] = ptr[0];
-					data[l++] = ptr[1];
-					data[l++] = ptr[2];
-					data[l++] = ptr[3];
+					data[l++] = num;
```
Comment (radare) [inline libr/asm/p/asm_x86_nz.c]: missing () & 0xFF ?

File: libr/asm/p/asm_x86_nz.c (hunk 1 above)
Comment (Svenito) [inline libr/asm/p/asm_x86_nz.c]: yes. I've seen `int` used in other places. Will replace with `bool` if preferable.

File: libr/asm/p/asm_x86_nz.c (hunk 2 above)
Comment (Svenito) [inline libr/asm/p/asm_x86_nz.c]: Not sure I understand. For what purpose?

Comment (radare) [inline libr/asm/p/asm_x86_nz.c]: 0xFF ensure there's no sign extension in the value which can produce wrong values when higher bit is set. But if it works in that way it should be fine i guess

Comment (Svenito) [inline libr/asm/p/asm_x86_nz.c]: for `mov [num], reg` if `num` is negative the opcode is fubar anyway compared to `nasm` output, and needs to be fixed. For positive values of `num` the output is fine. I'll be carrying on with this anyway.

File: libr/asm/p/asm_x86_nz.c [hunk 3]
```diff
@@ -1141,26 +1142,48 @@ SETNP/SETPO - Set if No Parity / Set if Parity Odd (386+)
...
 					delta = strchr (arg, '-');
 					if (delta) {
-						ut32 n = -getnum (a, delta + 1);
-						ut8 *N = (ut8 *)&n;
+						int r = getreg (arg);
+						int r2;
+						bool r2_is_num = false;
+						if (isnum (a, arg2)) {
+							r2 = getnum (a, arg2);
+							r2_is_num = 1;
```
Comment (radare) [inline libr/asm/p/asm_x86_nz.c]: if its bool, use true instead of 1 :P

**Comment by radare** [comment]
thx

**Pull request merged**